drivers/pwm/pwm_mcux.c

```diff
@@ -135,23 +141,33 @@ static int pwm_mcux_init(const struct device *dev)
 	struct pwm_mcux_data *data = dev->data;
 	pwm_config_t pwm_config;
 	status_t status;
-	int i, err;
+	int i;
 
 	if (!device_is_ready(config->clock_dev)) {
 		LOG_ERR("clock control device not ready");
 		return -ENODEV;
 	}
 
-	err = pinctrl_apply_state(config->pincfg, PINCTRL_STATE_DEFAULT);
+#ifdef CONFIG_PINCTRL
```

Comment:
Regarding the pin control changes: These should either be in a separate commit, or the commit message "drivers: pwm: lpc: Enabled Wait-Enable Bit required for LPC55s36 FlexPWM" should be updated to reflect that you are adding WAITEN bit support as well as making pin control optional for this driver.